pkg/sirenia/state/state.go

```diff
@@ -147,15 +147,20 @@ func (x *PgConfig) IsNewDownstream(y *PgConfig) bool {
 	return y.Downstream != nil && !peersEqual(x.Downstream, y.Downstream)
 }
 
-type Postgres interface {
+type Database interface {
+	// TODO(jpg): need a more generic representation of database log position.
+	// the only important thing here is to ensure we have an ordering.
+	// the comparison of this should also be handed of to the underlying database wrapper
 	XLogPosition() (xlog.Position, error)
-	Reconfigure(*PgConfig) error
+	// Returns the implementation of the XLog interface for this database
+	XLog() xlog.XLog
```

Comment:
I tried to eliminate the singleton but `Zero` is also used in the state machine to do various comparisons with the initial position. I think what we have now is actually as minimal as it can be barring Go tricks that I don't know yet.